### PR TITLE
codegen x86-64: Various improvements

### PR DESCRIPTION
**Pull request by PromyLOPh**
I ran into multiple issues wrt test case generation while adding new instructions. This commit request fixes them for me.

### CODE REVIEW

**Comment by Maratyszcza** [comment]
Changes to `codegen/x86_64_test_encoding.py` LGTM, but do not modify `setup.py`. Most Linux distributions do not have upstream `binutils` versions to generate these tests, which is why I keep pre-generated encoding tests inside the repo. I do not expect the users to re-generate tests themselves when they install PeachPy.

File: codegen/x86_64_test_encoding.py [hunk 1]
```diff
@@ -67,8 +69,8 @@ def binutils_encode(assembly):
         objcopy("-O", "binary", "-j", ".text", obj_file.name, bin_file.name)
         os.remove(obj_file.name)
     except OSError:
-        print(assembly)
-        raise
+        print('binutils failed for %s, cannot test' % assembly)
```
Comment (Maratyszcza) [inline codegen/x86_64_test_encoding.py]: Please revert. `binutils` encoding errors should be fatal.

**Comment by PromyLOPh** [comment]
Generating tests is still optional for normal builds. With this change
you can manually create them with `python setup.py generate
--with-tests` or use the developer target. (We could disable the latter,
arguing most people don’t want to add new instructions anyway.)

Tests are skipped if the assembler failed, because the binutils I’m
using right now don’t know about vpopcnt:

	Error: no such instruction: `vpopcntd zmm9{k6}{z},ZMMWORD PTR[r9+rbx*8 - 66]

That means I can’t test instructions I’m adding without upgrading my
binutils, even though the existing instructions are probably just fine
and don’t need to be tested by myself. However self.skipTest is probably
not the right choice. Suggestion:

	code.line("self.assertIsNotNone(%s, 'reference missing')" % reference_bytecode)

For now I’m reverting setup.py and binutils failure behavior.

**Comment by Maratyszcza** [comment]
Thank you, merged manually with minor changes as 962ac751967c37e4de7a1bc6cb5e578f3d279fc4.

**Pull request closed without merging**